packages/faceted-search/src/components/AddFacetPopover/AddFacetPopover.component.js

```diff
@@ -37,55 +69,189 @@ AddFacetRow.propTypes = {
 	label: PropTypes.string.isRequired,
 	onClick: PropTypes.func.isRequired,
+	isFocusable: PropTypes.bool.isRequired,
+};
+
+const AddFacetPopoverHeader = ({ category, onCategoryChange, id, resetFilter, onFilter, filterValue, t, isFocusable }) => (
+	<RichLayout.Header className={theme('tc-add-facet-popover-header')} id={`${id}-header`}>
+		{!isNull(category) && (
+			<div className={theme('tc-add-facet-popover-category')}>
+				<Button
+					bsStyle="link"
+					onClick={() => onCategoryChange(null)}
+					role="button"
+					className={theme('tc-add-facet-popover-category-btn')}
+					tabIndex={getTabIndex(isFocusable)}
+				>
+					<Icon name="talend-arrow-left" />
+				</Button>
+				<span className={theme('tc-add-facet-popover-category-title')}>{category}</span>
+			</div>
+		)}
+		<FilterBar
+			autoFocus={false}
+			className={theme('tc-add-facet-popover-filter')}
+			dockable={false}
+			docked={false}
+			iconAlwaysVisible
+			id={`${id}-filter`}
+			placeholder={t('ADD_FACET_FILTER_PLACEHOLDER', {
+				defaultValue: 'Find a filter',
+			})}
+			onToggle={resetFilter}
+			onFilter={onFilter}
+			value={filterValue}
+			tabIndex={getTabIndex(isFocusable)}
+		/>
+	</RichLayout.Header>
+);
+
+AddFacetPopoverHeader.propTypes = {
+	category: PropTypes.string,
+	onCategoryChange: PropTypes.func.isRequired,
+	id: PropTypes.string.isRequired,
+	resetFilter: PropTypes.func.isRequired,
+	onFilter: PropTypes.func.isRequired,
+	filterValue: PropTypes.string.isRequired,
+	isFocusable: PropTypes.bool.isRequired,
+	t: PropTypes.func.isRequired,
+};
+
+const filterByLabel = label => row => {
+	const rowLabel = isString(row) ? row : row.properties.label;
+
+	return rowLabel.toLowerCase().includes(label);
 };
 
-const filterByLabel = label => badgeDefinition =>
-	badgeDefinition.properties.label.toLowerCase().includes(label);
+const sortByLabel = (rowA, rowB) => {
+	const labelA = isString(rowA) ? rowA : rowA.properties.label;
+	const labelB = isString(rowB) ? rowB : rowB.properties.label;
 
-const sortByLabel = (badgeDefinitionA, badgeDefinitionB) =>
-	badgeDefinitionA.properties.label.localeCompare(badgeDefinitionB.properties.label);
+	return labelA.localeCompare(labelB);
+};
 
 const AddFacetPopover = ({ badgesDefinitions = [], id, initialFilterValue, onClick, t }) => {
-	const [filterValue, setFilterValue] = useState(initialFilterValue || '');
+	const addFacetId = `${id}-add-facet-popover`;
+
+	const [category, setCategory] = useState(null);
+	const [filterValue, setFilterValue] = useState(initialFilterValue || '')
+
+	const categories = useMemo(
+		() =>
+			badgesDefinitions
+				.filter(badgeDefinition => !!badgeDefinition.metadata.category)
+				.map(badgeDefinition => badgeDefinition.metadata.category)
+				.filter((categoryName, index, arr) => arr.indexOf(categoryName) === index),
+		[badgesDefinitions],
+	);
+
+	const screensCount = categories.length + 1;
+	const [screensHeight, setScreensHeight] = useState(times(screensCount, () => 0));
```

Comment:
we can use https://lodash.com/docs/4.17.15#constant
```suggestion
        import constant from 'lodash/constant';
	const [screensHeight, setScreensHeight] = useState(times(screensCount, constant(0)));
```